src/append_only_data.rs

```diff
@@ -176,37 +215,44 @@ impl Perm for UnpubPermissions {
         }
     }
 
+    /// Returns the last entry index.
     fn entries_index(&self) -> u64 {
         self.entries_index
     }
 
+    /// Returns the last owners index.
     fn owners_index(&self) -> u64 {
         self.owners_index
     }
 }
 
+/// Published permissions.
 #[derive(Clone, Serialize, Deserialize, PartialEq, PartialOrd, Ord, Eq, Hash, Debug)]
 pub struct PubPermissions {
+    /// Map of users to their published permission set.
     pub permissions: BTreeMap<User, PubPermissionSet>,
-    /// The current index of the data when this permission change happened
+    /// The current index of the data when this permission change happened.
     pub entries_index: u64,
-    /// The current index of the owners when this permission change happened
+    /// The current index of the owners when this permission change happened.
     pub owners_index: u64,
 }
 
 impl PubPermissions {
+    /// Returns true if `action` is allowed for the provided user.
     fn is_action_allowed_by_user(&self, user: &User, action: Action) -> Option<bool> {
         self.permissions
             .get(user)
             .and_then(|perms| perms.is_allowed(action))
     }
 
+    /// Gets the complete list of permissions.
     pub fn permissions(&self) -> &BTreeMap<User, PubPermissionSet> {
         &self.permissions
     }
 }
 
 impl Perm for PubPermissions {
+    /// Returns true if `action` is allowed for the provided user.
```

Comment:
This function doesn't return a boolean, so should be
```suggestion
    /// Returns `Ok(())` if `action` is allowed for the provided user and `Err(AccessDenied)` if
    /// this action is not permitted.
```